### ISSUE TITLE
Generate Soap Client rather than make it a dependency

### ISSUE THREAD

**Issue by sanbornm** [hooklift/gowsdl#36]
I love what has been built so far, one suggestion:

I feel like the common use case for gowsdl is to be used as a one time code generator.  I noticed there were some open issues regarding authentication and other client related features requested.  I personally need several types of authentication as well as proxy support.  I can easily add these with a few lines of code if I am in control of the soap client.

I think gowsdl should be responsible for only code generation and not function as a library.  The soap client should be generated along with the types and allow the user to modify it to fit their needs.  This has some benefits:

* Generated code wouldn't need to depend on an external library
* Gowsdl could be forked and built in another repo as it wouldn't have embedded dependency paths.  Currently you have to `go get` the main repo and adjust your git remote in order to work on it.
* This frees gowsdl from trying to support every possible auth/header combination and allows the project to focus on getting the type generation correct (it currently doesn't support sequence>choice>element types or navigate around duplicate type names and I am sure other things :smiley:  ).

In summary, I think gowsdl should inject the soap client into the generated code, require the user to have no external dependencies, and focus on getting correct compilable code.  I am willing to restructure the output of the soap client (and add tests) if this is something that you would consider merging in.

Thoughts?

**Comment by c4milo** [comment]
Hey @sanbornm, I'm not against that idea at all.

**Comment by c4milo** [comment]
This landed in https://github.com/hooklift/gowsdl/commit/4e0b9fe0dc557f586dacca7cbdc4c2afe683524e
